homeassistant/components/calendar/__init__.py

```diff
@@ -176,10 +185,29 @@ def _get_date(date):
 
         # cleanup the string so we don't have a bunch of double+ spaces
         self._cal_data['message'] = re.sub('  +', '', summary).strip()
-
         self._cal_data['offset_time'] = offset_time
         self._cal_data['location'] = self.data.event.get('location', '')
         self._cal_data['description'] = self.data.event.get('description', '')
         self._cal_data['start'] = start
         self._cal_data['end'] = end
         self._cal_data['all_day'] = 'date' in self.data.event['start']
+
+
+class CalendarEventView(http.HomeAssistantView):
+    """View to retrieve calendar content."""
+
+    name = "api:calendar"
+
+    def __init__(self, device):
+        """Initialize calendar view."""
+        self.device = device
+        self.url = '/api/calendar/' + device.entity_id.split(".")[-1]
+
+    async def get(self, request):
+        """Return calendar events."""
+        if hasattr(self.device, "update"):
+            self.device.update()
+        elif hasattr(self.device, "async_update"):
+            await self.device.async_update()
+        _event_list = getattr(self.device.data, "event_list", [])
```

Comment:
This is still using a cached property.

All platforms need to have a new method added:

```
async def get_events(self, start_date, end_date)
```

This method should return the events for the given period. This should be computed by the platform on the fly, not using a cached property `event_list`.